### ISSUE TITLE
[BUG] data source connection link in Index pattern table breaks

### ISSUE THREAD

**Issue by zhongnansu** [opensearch-project/OpenSearch-Dashboards#6581]
**Describe the bug**

data source connection link in Index pattern table is not working as expected. It fails to redirect to the corresponding data source detail page, instead it goes to the dashboards management landing page

**Screenshots**

![iShot_2024-04-21_23 18 23](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/32652829/6320cdba-8f4b-472a-907b-e8c3ad57c6e2)

**Comment by zhongnansu** [comment]
I'll work on this

The issue is with the link url. Currently it's
```
/app/management/opensearch-dashboards/dataSourceManagement/<data source id>
```
it should be 

```
/app/management/opensearch-dashboards/dataSourceManagement/<data source id>
```
`dataSourceManagement` is the data source management plugin id

cc: @BionIT

**Comment by seraphjiang** [comment]
> The issue is with the link url. Currently it's
> 
> ```
> /app/management/opensearch-dashboards/dataSourceManagement/<data source id>
> ```
> 
> it should be
> 
> ```
> /app/management/opensearch-dashboards/dataSourceManagement/<data source id>
> ```
> 
> `dataSourceManagement` is the data source management plugin id

Hi @zhongnansu the actual and expected Url seems identical, would you check

**Comment by zhongnansu** [comment]
@seraphjiang sorry for typo, currently it's
```
/app/management/opensearch-dashboards/dataSources/<data source id>
```
E.g: https://future.playground.opensearch.org/app/management/opensearch-dashboards/dataSources/93d6c680-dcbd-11ee-987c-f557473dd2d5

**Comment by zhongnansu** [comment]
fixed by PR ##6604, closing
